purchase_receipt_progress/models/purchase_order.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderReceivedByCategory(models.Model):
+    _name = "purchase.order.received.by.category"
+    _description = "Purchase Order Received by Category"
+
+    order_id = fields.Many2one("purchase.order")
+    category_id = fields.Many2one("product.category")
```

Comment:
What happens when a category is deleted?

Reply:
it is not required, so presumably nothing (ondelete set null)